File: lib/components/forms/public_order_strings.dart
import 'package:easy_localization/easy_localization.dart';
import 'package:flutter/widgets.dart';

class PublicOrderStrings {
  static const String _prefix = 'PublicOrder';

  static String _tr(String key, String? tone) {
    if (tone == 'informal') {
      final informalKey = '${key}_informal';
      if (informalKey.tr() != informalKey) {
        return informalKey.tr();
      }
    }
    return key.tr();
  }

  // Re-verify the logic:
  // If I call _tr('PublicOrder.successTitle', 'informal')
  // It checks 'PublicOrder.successTitle_informal'.

  static String successTitle(String? tone, {bool hasTickets = true}) {
    if (hasTickets) {
      return _tr('$_prefix.successTitle', tone);
    }
    return _tr('$_prefix.successRegistrationTitle', tone);
  }

  static String paymentInfo(String? tone) => _tr('$_prefix.paymentInfo', tone);

  static String selectSeat(String? tone) => _tr('$_prefix.selectSeat', tone);

  static String getSubmitButton(bool hasTickets, String? tone) {
    if (hasTickets) {
      return _tr('$_prefix.submitOrder', tone);
    }
    return _tr('$_prefix.submitRegistration', tone);
  }

  static String get summary => '$_prefix.summary'.tr();
  static String get backToForm => '$_prefix.backToForm'.tr();
  static String get orderFailed => '$_prefix.orderFailed'.tr();

  static String orderError(String code) =>
      '$_prefix.orderError'.tr(namedArgs: {"code": code});

  static String productUnavailable(String productTitle) =>
      '$_prefix.productUnavailable'.tr(
        namedArgs: {"product_title": productTitle},
      );

  static String chooseDifferentVariant(String? tone) =>
      _tr('$_prefix.chooseDifferentVariant', tone);

  static String totalPrice(BuildContext context, String price) =>
      '$_prefix.totalPrice'.tr(namedArgs: {"price": price});

  static String get paymentMethod => '$_prefix.paymentMethod'.tr();
  static String get payFullAmount => '$_prefix.payFullAmount'.tr();
  static String get payWithDeposit => '$_prefix.payWithDeposit'.tr();
  static String get depositInfoOnSite => '$_prefix.depositInfoOnSite'.tr();
  static String depositInfoDaysBefore(int days) =>
      '$_prefix.depositInfoDaysBefore'.tr(namedArgs: {"days": days.toString()});

  // Payment QR (shown on the order confirmation when the form enables it).
  static String get paymentQrTitle => '$_prefix.paymentQrTitle'.tr();
  static String get paymentQrSubtitle => '$_prefix.paymentQrSubtitle'.tr();
  static String get amountToPay => '$_prefix.amountToPay'.tr();
  static String get bankAccount => '$_prefix.bankAccount'.tr();
  static String get iban => '$_prefix.iban'.tr();
  static String get variableSymbol => '$_prefix.variableSymbol'.tr();
  static String get paymentReference => '$_prefix.paymentReference'.tr();
  static String get paymentNote => '$_prefix.paymentNote'.tr();
  static String get orderLabel => '$_prefix.orderLabel'.tr();
  static String get downloadQr => '$_prefix.downloadQr'.tr();
  static String get copied => '$_prefix.copied'.tr();
}

File: web_client/src/components/forms/public_order_strings.js
import { LocalizationService } from '../../services/localization_service.js';

export class PublicOrderStrings {
    static _prefix = 'PublicOrder';

    static _tr(key, tone) {
        if (tone === 'informal') {
            const informalKey = `${key}_informal`;
            const val = LocalizationService.tr(informalKey);
            if (val && val !== informalKey) {
                return val;
            }
        }
        return LocalizationService.tr(key);
    }

    static successTitle(tone, hasTickets = true) {
        if (hasTickets) {
            return PublicOrderStrings._tr(`${PublicOrderStrings._prefix}.successTitle`, tone);
        }
        return PublicOrderStrings._tr(`${PublicOrderStrings._prefix}.successRegistrationTitle`, tone);
    }

    static paymentInfo(tone) {
        return PublicOrderStrings._tr(`${PublicOrderStrings._prefix}.paymentInfo`, tone);
    }

    static selectSeat(tone) {
        return PublicOrderStrings._tr(`${PublicOrderStrings._prefix}.selectSeat`, tone);
    }

    static getSubmitButton(hasTickets, tone) {
        if (hasTickets) {
            return PublicOrderStrings._tr(`${PublicOrderStrings._prefix}.submitOrder`, tone);
        }
        return PublicOrderStrings._tr(`${PublicOrderStrings._prefix}.submitRegistration`, tone);
    }

    static get summary() {
        return LocalizationService.tr(`${PublicOrderStrings._prefix}.summary`);
    }

    static get backToForm() {
        return LocalizationService.tr(`${PublicOrderStrings._prefix}.backToForm`);
    }

    static get orderFailed() {
        return LocalizationService.tr(`${PublicOrderStrings._prefix}.orderFailed`);
    }

    static orderError(code) {
        return LocalizationService.tr(`${PublicOrderStrings._prefix}.orderError`, { code: code });
    }

    static productUnavailable(productTitle) {
        return LocalizationService.tr(`${PublicOrderStrings._prefix}.productUnavailable`, { product_title: productTitle });
    }

    static chooseDifferentVariant(tone) {
        return PublicOrderStrings._tr(`${PublicOrderStrings._prefix}.chooseDifferentVariant`, tone);
    }

    static totalPrice(price) {
        return LocalizationService.tr(`${PublicOrderStrings._prefix}.totalPrice`, { price: price });
    }
}
